spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/main/java/org/springframework/boot/configurationprocessor/fieldvalues/javac/JavaCompilerFieldValuesParser.java

```diff
@@ -176,10 +176,10 @@ private Object getValue(ExpressionTree expression, Object defaultValue) throws E
 				}
 				return result;
 			}
-			if (expression.getKind().equals("IDENTIFIER")) {
+			if ("IDENTIFIER".equals(expression.getKind())) {
```

Comment:
Those two aren't necessary as the kind of a tree is a mandatory element of the expression.